src/daft-io/src/python.rs

```diff
@@ -66,10 +66,40 @@ mod py {
         Ok(crate::set_io_pool_num_threads(num_threads as usize))
     }
 
+    #[pyfunction]
+    fn io_config_from_env(py: Python) -> PyResult<common_io_config::python::IOConfig> {
+        let io_config: DaftResult<common_io_config::IOConfig> = py.allow_threads(|| {
+            let runtime = get_runtime(false)?;
+            let runtime_handle = runtime.handle();
+            let _rt_guard = runtime_handle.enter();
+            runtime_handle.block_on(async {
+                Ok(common_io_config::IOConfig {
+                    s3: s3_like::s3_config_from_env().await?,
+                    // TODO(jay): Derive credentials for GCS and Azure
+                    gcs: common_io_config::GCSConfig::default(),
+                    azure: common_io_config::AzureConfig::default(),
+                })
+            })
+        });
+        Ok(io_config?.into())
+    }
+
     pub fn register_modules(py: Python, parent: &PyModule) -> PyResult<()> {
         common_io_config::python::register_modules(py, parent)?;
+
         parent.add_function(wrap_pyfunction!(io_glob, parent)?)?;
         parent.add_function(wrap_pyfunction!(set_io_pool_num_threads, parent)?)?;
+        parent.add_function(wrap_pyfunction!(io_config_from_env, parent)?)?;
+
+        // Patch the IOConfig Python class, which was registered with common_io_config::python::register_modules,
+        // to have a new `IOConfig.from_env` constructor
+        let io_config_from_env_pyfunc = parent
+            .getattr("io_config_from_env")
+            .expect("io_config_from_env should be registered");
+        parent
+            .getattr("IOConfig")
+            .expect("IOConfig class should be registered")
+            .setattr("from_env", io_config_from_env_pyfunc)?;
```

Comment:
This is the jankiest part of the PR:

* I didn't want to implement `IOConfig.from_env` directly on the `common::io_config` crate because I wanted to avoid a dependency on the AWS SDK for our common/config crates
* However, the `IOConfig` class is defined and registered from `common::io_config::python`
* Hence I decided to "patch" the `IOConfig.from_env` constructor on during the registration of this `daft-io` crate

Not sure if there is a better solution, open to suggestions!

Reply:
Fixed by moving this to a dynamic Python import in `S3Config.from_env`